### ISSUE TITLE
Emit PossiblyInvalidMethodCall if there are other types in the union type which define the method?

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#299]
Related to https://github.com/vimeo/psalm/issues/283

Reason for this: `InvalidMethodCall` would have a much lower false positive rate than `PossiblyInvalidMethodCall` if those were split up. Psalm already has PossiblyFalse and PossiblyNull for method calls.

```php
<?php

class A1 {
    public function methodOfA(): void {
    }   
}

/** @param A1|string $x */
function example($x, bool $isObject) {
    if ($isObject) {
        $x->methodOfA();  // Observed error type: InvalidMethodCall. Desired: PossiblyInvalidMethodCall
    }   
}
```

Output:

```
ERROR: InvalidMethodCall - src/method.php:11:9 - Cannot call method methodOfA on A1|string variable $x
        $x->methodOfA();  // Observed error type: InvalidMethodCall. Desired: PossiblyInvalidMethodCall
```

**Comment by muglug** [comment]
I'm not entirely sure these false positive rates would be much reduced – I think that the vast majority of `InvalidMethodCall` issues are on union types with just one atomic type. Nevertheless, I'll add for completeness.
